impl/src/main/java/dao/TagDao.java

```diff
@@ -0,0 +1,30 @@
+package dao;
+
+import api.Tag;
+import rx.Observable;
+import se.fortnox.reactivewizard.db.GeneratedKey;
+import se.fortnox.reactivewizard.db.Query;
+import se.fortnox.reactivewizard.db.Update;
+
+import java.util.List;
+
+public interface TagDao {
+
+    @Query(value = "SELECT label FROM tag WHERE label ILIKE ( :search || '%')")
+    Observable<Tag> getTagsBySearchQuery(String search);
+
+    @Query("SELECT id, label FROM tag WHERE label IN (:labels)")
+    Observable<Tag> getTagsByLabels(List<String> labels);
+
+    @Update("INSERT INTO tag (label) VALUES (:label) RETURNING id, label")
+    Observable<GeneratedKey<Tag>> createTag(String label);
```

Comment:
Any restrictions on label? Should we allow spaces, special chars etc?

Reply:
Yup, we have reactivewizard-validation available so I'll align rules on Tag with our frontend.